wirecard/gambio-ee
# WDG-63: Code Style Automation: PHP

### This PR

* Adds PHPCS to check for PHP coding guidelines
* Adds both plugin-independent and plugin-specific rules/sniffs
* Adds Rake tasks to run PHPCS and PHPCS Fixer

### Notes

* The rules are taken from:
  * The [PSR-2 standard](https://www.php-fig.org/psr/psr-2/)
  * The [common Wirecard coding guidelines](https://bitbucket.org/parksideit/wirecard-documentation/src/master/#markdown-header-automatically-checked)
  * Gambio's [core Coding Guidelines](https://developers.gambio.de/tutorials.html?v=3.13.1.0&p=guidelines)
  * Gambio's [PHP Coding Guidelines](https://developers.gambio.de/tutorials.html?v=3.13.1.0&p=php-guidelines)

### How to Test

* Run `rake cs_check` to check for errors/warnings
* Run `rake cs_fix` to automatically fix errors/warnings

### Jira Links

* WDG-63: https://jira.parkside.at/browse/WDG-63
